python/libraries/bimvee/geometry.py

```diff
@@ -101,6 +101,18 @@ def pose6qInterp(poseDict, time):
     locOut = locPre * (1-timeRel) + locPost * timeRel
     return (locOut, qOut)
 
+def combineTwoQuaternions(q1, q2):
+    w1 = q1[0]
+    v1 = q1[1:4]
+    w2 = q2[0]
+    v2 = q2[1:4]
+    wOut = w1*w2 - np.dot(v1.T, v2)
+    vOut = w1*v2 + w2*v1 + np.dot(v1, v2)
+    qOut = np.zeros_like(q1)
+    qOut[0] = wOut
+    qOut[1:4] = vOut
+    return qOut
+
```

Comment:
The output quaternion here is equivalent to the rotation by quaternion q1, followed by rotation by quaternion q2, right? If so, then the output seems incorrect. It should be equivalent to the quaternion product (q2 . q1). I called it with some different rotations (unit quaternions) and the superposition principle doesn't hold for the outputted quaternion.

Reply:
Good catch - I've corrected that.

Reply:
(dot product instead of cross product).